### PR TITLE
[V3.6.0 pipeline] added web forward and deferred pipelines

### PR DESCRIPTION
**Pull request by star-e**
web-forward and web-deferred pipelines are added.

Has no impact on current pipeline.

### Changelog

* added getMacroXXX to PipelineRuntime
* updated render-config.json to enable feature cropping

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/core/renderer/core/program-lib.ts [hunk 1]
```diff
@@ -443,28 +443,9 @@ class ProgramLib {
...
-                }
-            } else {
-                this.getDescriptorSetLayout(device, name); // ensure set layouts have been created
-                insertBuiltinBindings(tmpl, tmplInfo, globalDescriptorSetLayout, 'globals');
-                tmplInfo.setLayouts[SetIndex.GLOBAL] = pipeline.descriptorSetLayout;
-            }
+            this.getDescriptorSetLayout(device, name); // ensure set layouts have been created
+            insertBuiltinBindings(tmpl, tmplInfo, globalDescriptorSetLayout, 'globals');
+            tmplInfo.setLayouts[SetIndex.GLOBAL] = pipeline.descriptorSetLayout;
             tmplInfo.pipelineLayout = device.createPipelineLayout(new PipelineLayoutInfo(tmplInfo.setLayouts));
```
Comment (star-e) [inline cocos/core/renderer/core/program-lib.ts:449]: reverted previous changes of custom pipeline

File: cocos/core/pipeline/pipeline-ubo.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ import { UBOGlobal, UBOShadow, UBOCamera, UNIFORM_SHADOWMAP_BINDING,
 import { Device, BufferInfo, BufferUsageBit, MemoryUsageBit, DescriptorSet, API } from '../gfx';
 import { Camera } from '../renderer/scene/camera';
 import { Mat4, Vec3, Vec4, Color } from '../math';
-import { RenderPipeline } from './render-pipeline';
+import { PipelineRuntime } from './custom/pipeline';
```
Comment (star-e) [inline cocos/core/pipeline/pipeline-ubo.ts:31]: unify Pipeline interface

File: editor/engine-features/render-config.json [hunk 3]
```diff
@@ -222,6 +222,12 @@
             "description": "i18n:ENGINE.features.marionette.description",
             "wechatPlugin": false,
             "category": "animation"
+        },
+        "custom-pipeline": {
+            "default": ["custom-pipeline"],
+            "label": "i18n:ENGINE.features.custom_pipeline.label",
+            "description": "i18n:ENGINE.features.custom_pipeline.description",
+            "wechatPlugin": false
```
Comment (star-e) [inline editor/engine-features/render-config.json:230]: added new feature, editor need to enable this

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**